test/core/adt/array_test.cpp

```diff
@@ -1,104 +0,0 @@
-/**
```

Comment:
Why to delete all tests?

Reply:
They became irrelevant, as multimap is the only user of "append", and multimap tests cover that behavior.